### ISSUE TITLE
Crash When Hovering Filled ShulkerBoxes

### ISSUE THREAD

**Issue by ardissaps** [ShetiPhian/Localizations-and-Issues#362]
It says unknown TooltipComponent :
 https://paste.atlauncher.com/view/38f8c98f-3cba-4ebf-badd-e3c725db40f4

Possible Conflict (only guessing) : 
[EasyShulkerBoxes](https://www.curseforge.com/minecraft/mc-mods/easy-shulker-boxes)

Edit : It is indeed incompatible with that. I tested both work but can't run together which cause crash when hovering shulkerbox with content inside it.

**Comment by ShetiPhian** [comment]
Like vanilla my tooltip processer only handles TooltipData that is an instance of BundleTooltipData, and crashes if its not. This is because I'm also using TooltipComponent.of(TooltipData).

Since my code lacks the additional edits made to vanilla to handle the custom TooltipData and stop TooltipComponent.of from processing it, the game crashes.

A few hours ago I posted a test build in the related issue: https://github.com/ShetiPhian/Localizations-and-Issues/issues/360
https://www.mediafire.com/file/19skdt2vd5590tz/shetiphiancore-fabric-1.18-1.2.13--TESTER.jar/file

**Comment by ardissaps** [comment]
That file does not crash but disabled the shulker preview on hover from EasyShulkerBox.

**Comment by ardissaps** [comment]
Now I am replacing EasyShulkerBox with Tweakeroo feature, which works with your mods. And the right click insert item using Charm feature.

**Comment by ShetiPhian** [comment]
Not surprised the preview stopped working, but now that I know of a mod to test with I can try out some ideas to either improve support or fall back as if my code didn't exist.

**Comment by ardissaps** [comment]
> Not surprised the preview stopped working, but now that I know of a mod to test with I can try out some ideas to either improve support or fall back as if my code didn't exist.

I decided to try again because I really like how EasyShulkerBox render tooltip, and still using shetipiancore with wrapped thing disabled and it now works fine. If it is enable to make tooltip not going to the edge, I think most people are using [TooltipFix](https://modrinth.com/mod/tooltipfix) already instead or [Debugify](https://modrinth.com/mod/debugify) which contain part of the feature from tooltipfix. Consider make it disabled by default? Is it critical to enable that?

**Comment by Fuzss** [comment]
Hey there, why don't you simply call [the Fabric event](https://github.com/FabricMC/fabric/blob/1.19/fabric-rendering-v1/src/client/java/net/fabricmc/fabric/api/client/rendering/v1/TooltipComponentCallback.java) for mods to provide their own `ClientTooltipComponent` when you create those from their respective `TooltipComponent`? You have Fabric API as a dependency already, so why not use it?

**Comment by ShetiPhian** [comment]
https://www.mediafire.com/file/8lb2fxmvzb2q8az/shetiphiancore-fabric-1.18-1.2.13--TESTER3.jar/file
This should solve the issue (EasyShulkerBoxes works)


About the event; I didn't know of it, and there wasn't an indication I needed to look for it.
Tooltips where broken on Fabric but not Forge, so I looked where it was fixed and implemented similar code.
Both Forge and Vanilla checked if the Optional was present and used the same method to convert it, so I did the same.

In the end I switched to hooking the streamed map. No longer touching TooltipData, or converting to TooltipComponents, or calling the renderer. Just dealing with a list of Text and returning one of OrderedText. The wrapping code is cleaner too.


Why include this? I simply want my stuff to work out of the box.
The fewer required third party libraries the better, Myron is getting tosses asap, its caused enough annoyances.
